packages/gatsby-plugin-manifest/src/gatsby-browser.js

```diff
@@ -0,0 +1,26 @@
+import { withPrefix as fallbackWithPrefix, withAssetPrefix } from "gatsby"
+
+const withPrefix = withAssetPrefix || fallbackWithPrefix
+
+exports.onRouteUpdate = function({ location }, pluginOptions) {
+  const { localize } = pluginOptions
+  const manifestFilename = getManifestForPathname(location.pathname, localize)
+
+  const manifestEl = document.head.querySelector(`link[rel="manifest"]`)
+  if (manifestEl) {
+    manifestEl.setAttribute(`href`, withPrefix(manifestFilename))
+  }
+}
+
+function getManifestForPathname(pathname, localizedApps) {
+  let suffix = ``
+  if (Array.isArray(localizedApps)) {
+    const appOptions = localizedApps.find(app =>
+      RegExp(`^${app.start_url}.*`, `i`).test(pathname)
```

Comment:
I haven't tested this regex but just crossed my mind that we need to be super careful with string matching. For instance, there's a valid country code 'an'. this could erroneously be detected if it detects the `m(an)ifest.webmanifest`. Like I said, not sure if it does but we should do some thorough testing.

Reply:
i'm guessing this one could be an issue:

```
Type: language
Subtag: st
Description: Southern Sotho
Added: 2005-10-16
Suppress-Script: Latn
```

Reply:
I am way to tired to know if these last two comments are genius or rubbish. I have a feeling their rubbish and it's not going to be a problem...please read and disregard or fix as necessary. I'll re read when I'm not wiped. :sob:

Reply:
Hahah, get some rest!

This regex is the same as the one already used in `gatsby-ssr.js`: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-manifest/src/gatsby-ssr.js#L22

I might be misunderstanding what you meant, but this line doesn't have anything to do with country codes or the name of the manifest file. The line checks to see if the path of the URL (e.g. `/es/`) matches the `start_url` specified in the localized app's manifest.